File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        stack = []
        dic={')':'(', '}':'{',']':'['}
        for i in s:
            if stack and (i in dic and stack[-1] == dic[i]):
                stack.pop()
            else:
                stack.append(i)
        return not stack

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack=new Stack<>();
        Map<Character,Character> map=new HashMap<>();
        map.put(')','(');
        map.put('}','{');
        map.put(']','[');

        for(char c : s.toCharArray()) {
            if(!stack.isEmpty() && map.containsKey(c) && stack.peek() == map.get(c)){
                stack.pop();
            } else {
                stack.push(c);
            }
        }
        return stack.isEmpty();
    }
}